CMakeLists.txt

```diff
@@ -115,7 +115,12 @@ if (APPLE)
     endif()
 endif()
 
-set(NGRAPH_CXX_STANDARD 11)
+if (NGRAPH_MLIR_ENABLE)
+  #MLIR uses the latest version of LLVM which requires C++14.
+  set(NGRAPH_CXX_STANDARD 14)
+else()
+  set(NGRAPH_CXX_STANDARD 11)
+endif()
```

Comment:
I found doing this easier than using LLVM_CXX_STD since LLVM cmake hasn't been imported at this point and that var doesn't exist in the the LLVM version we use without MLIR